content/chainguard/migration/dockerfile-conversion.md

```
@@ -0,0 +1,233 @@
+
+
+For each `FROM` line in the Dockerfile, `dfc` attempts to replace the base image with an equivalent Chainguard Image. For each `RUN` line in the Dockerfile, `dfc` attempts to detect the use of a known package manager (e.g. `apt` / `yum` / `apk`) and extracts the names of any packages being installed. It then attempts to map these packages to Chainguard equivalent APKs, defaulting to the existing names. All existing package manager commands are then removed and replaced with a single `apk add -U <packages>`. Additionally, dfc will add a `USER root` instruction to allow package installations since most Chainguard Images run as a regular, non-root user.
+
+{{< note >}}
+dfc is an open source tool that is still under active development and subject to changes. While we try to cover a variety of use case scenarios, some inconsistencies may occur due to the diverse nature of Dockerfiles and package manager instructions. There may be errors or gaps in functionality as you use the feature in the early access phase. Please let us know if you come across any issues or have any questions.You can [file an issue](https://github.com/chainguard-dev/dfc/issues/new/choose) on GitHub to get in touch.
+{{< /note >}}
+
+## Installation
+
+You’ll need a Go environment to install and run Chainguard’s dfc tool. To install it on your local system, run:
+
+```
+go install github.com/chainguard-dev/dfc@latest
+```
+
+To verify that the installation was successful, run:
+
+```
+dfc --help
+```
+
+You will receive output with basic usage instructions.
+
+```
+Usage:
+  dfc [flags]
+
+Examples:
+dfc <path_to_dockerfile>
+
+Flags:
+  -h, --help         help for dfc
+  -i, --in-place     modified the Dockerfile in place (vs. stdout), saving original in a .bak file
+      --json         print dockerfile as json (pre-conversion)
+      --org string   the root repo namespace at cgr.dev/<org> (default "ORGANIZATION")
+```
+
+## Basic Usage
+
+Unless specified, dfc will not make any direct changes to your Dockerfile, writing the results to the default output stream. Run the following to convert a Dockerfile and save the output to a new file:
+
+```
+dfc ./Dockerfile > ./Dockerfile.converted
+```
+
+You can also pipe the Dockerfile’s contents from stdin:
+
+```
+cat ./Dockerfile | dfc -
+```
+
+The following CLI flags are available:
+
+* `--org=<org>` \- the registry namespace, i.e. `cgr.dev/<org>` (default placeholder: `ORGANIZATION`)
+* `--json` \- serialize Dockerfile as JSON
+* `--in-place / -i` \- modify the Dockerfile in place vs. printing to stdout, saving original in a .bak file
+
+## Examples
+
+This section has a few practical examples you can use as reference.
+
+### Setting the ORG
+
+By default, `dfc` uses `ORGANIZATION` as a placeholder for the image registry address. You can provide the `--org` parameter to specify the organization that you’re a member of. To use free tier images, use `chainguard` as the organization.
```

```suggestion
By default, `dfc` uses `ORG` as a placeholder for the image registry address. You can provide the `--org` parameter to specify the organization that you’re a member of. To use free tier images, use `chainguard` as the organization.
```

On my run through it seems like `ORG` is the placeholder